src/test/Drips.t.sol

```diff
@@ -106,6 +106,82 @@ contract DripsTest is DSTest {
         return recv(recv(recv1, recv2, recv3), recv4);
     }
 
+    function random(bytes32 name, uint256 salt) internal pure returns (uint256) {
+        return random(name, salt, type(uint256).max);
+    }
+
+    function random(
+        bytes32 name,
+        uint256 salt,
+        uint256 range
+    ) internal pure returns (uint256) {
+        return uint256(keccak256(abi.encode(name, salt))) % range;
+    }
+
+    function combineSalt(uint256 a, uint256 b) public pure returns (uint256) {
+        return uint256(keccak256(abi.encode(a, b)));
+    }
+
+    function genRandomRecv(
+        uint256 randomSalt,
+        uint8 amountReceiver,
+        uint128 maxAmtPerSec,
+        uint32 maxStart,
+        uint32 maxDuration
+    ) internal pure returns (DripsReceiver[] memory) {
+        uint256 inPercent = 100;
+        uint256 probDefaultEnd = random("default.end", randomSalt, inPercent);
+        uint256 probStartNow = random("start.now", randomSalt, inPercent);
+        DripsReceiver[] memory receivers;
+        for (uint8 i = 0; i < amountReceiver; i++) {
+            uint256 salt = combineSalt(randomSalt, i);
+            uint256 amtPerSec = random("amtPerSec", salt, maxAmtPerSec) + 1;
+            uint256 start = random("start", salt, maxStart);
+            if (start % 100 <= probStartNow) start = 0;
+            uint256 duration = random("duration", salt, maxDuration);
+            if (duration % 100 <= probDefaultEnd) duration = 0;
+
+            receivers = recv(receivers, recv(uint256(i), amtPerSec, start, duration));
+        }
+        return receivers;
+    }
+
+    function receiveDrips(
+        DripsReceiver[] memory receivers,
+        uint32 defaultEnd,
+        uint32 updateTime
+    ) internal {
+        emit log_named_uint("defaultEnd:", defaultEnd);
+        for (uint256 i = 0; i < receivers.length; i++) {
+            DripsReceiver memory r = receivers[i];
+            uint32 duration = r.duration;
+            uint32 start = r.start;
+            if (start == 0) start = updateTime;
+            if (duration == 0) duration = defaultEnd - start;
+            if (start < updateTime) duration -= updateTime - start;
```

Comment:
I tried to do these checks a bit differently compared with the dripsRange function.

Obviously,  we could have also called dripsRange function here but I think it is good to do it a bit different to catch potential errors in dripsRange function.

Reply:
I will run it with multiple fuzz runs tonight. Just in case I forgot one of the cases here.